Mathlib/Data/Multiset/Antidiagonal.lean

```diff
@@ -0,0 +1,127 @@
+about the parentheses being used instead of `<|` than the typo. -/
+/- Porting note: `simp` seemed to be applying `antidiagonal_coe'` instead of `antidiagonal_coe`
+in what used to be `simp [antidiagonal_coe]`. -/
+/-- A pair `(t₁, t₂)` of multisets is contained in `antidiagonal s`
+    if and only if `t₁ + t₂ = s`. -/
+@[simp]
+theorem mem_antidiagonal {s : Multiset α} {x : Multiset α × Multiset α} :
+    x ∈ antidiagonal s ↔ x.1 + x.2 = s :=
+  Quotient.inductionOn s <| fun l ↦ by
+    dsimp [antidiagonal_coe]; refine' ⟨fun h => revzip_powersetAux h, fun h ↦ _⟩
+    haveI := Classical.decEq α
+    simp [revzip_powersetAux_lemma l revzip_powersetAux, h.symm]
+    cases' x with x₁ x₂
+    dsimp only
+    exact ⟨x₁, le_add_right _ _, by rw [add_tsub_cancel_left x₁ x₂]⟩
+#align multiset.mem_antidiagonal Multiset.mem_antidiagonal
+
+@[simp]
+theorem antidiagonal_map_fst (s : Multiset α) : (antidiagonal s).map Prod.fst = powerset s :=
+  Quotient.inductionOn s <| fun l ↦ by simp [powersetAux'];
+#align multiset.antidiagonal_map_fst Multiset.antidiagonal_map_fst
+
+@[simp]
+theorem antidiagonal_map_snd (s : Multiset α) : (antidiagonal s).map Prod.snd = powerset s :=
+  Quotient.inductionOn s <| fun l ↦ by simp [powersetAux']
+#align multiset.antidiagonal_map_snd Multiset.antidiagonal_map_snd
+
+/- Porting note: I changed `@antidiagonal` to `@antidiagonal.{u_1}` because otherwise I got
+`Multiset.antidiagonal_zero.{u_2, u_1} {α : Type (max u_1 u_2)} : ...`, which gave me issues
+and triggered the linter. -/
+@[simp]
+theorem antidiagonal_zero : @antidiagonal.{u_1} α 0 = {(0, 0)} :=
+  rfl
+#align multiset.antidiagonal_zero Multiset.antidiagonal_zero
+
+@[simp]
+theorem antidiagonal_cons (a : α) (s) :
+    antidiagonal (a ::ₘ s) =
+      map (Prod.map id (cons a)) (antidiagonal s) + map (Prod.map (cons a) id) (antidiagonal s) :=
+  Quotient.inductionOn s <| fun l ↦
+    by
+    simp only [revzip, reverse_append, quot_mk_to_coe, coe_eq_coe, powersetAux'_cons, cons_coe,
+      coe_map, antidiagonal_coe', coe_add]
+    rw [← zip_map, ← zip_map, zip_append, (_ : _ ++ _ = _)]
+    · congr ; simp ; rw [map_reverse] ; simp
+    · simp
+#align multiset.antidiagonal_cons Multiset.antidiagonal_cons
+
+theorem antidiagonal_eq_map_powerset [DecidableEq α] (s : Multiset α) :
+    s.antidiagonal = s.powerset.map fun t ↦ (s - t, t) :=
+  by
+  induction' s using Multiset.induction_on with a s hs
+  · simp only [antidiagonal_zero, powerset_zero, zero_tsub, map_singleton]
+  · simp_rw [antidiagonal_cons, powerset_cons, map_add, hs, map_map, Function.comp, Prod.map_mk,
+      id.def, sub_cons, erase_cons_head]
+    rw [add_comm]
+    congr 1
+    refine' Multiset.map_congr rfl fun x hx ↦ _
+    rw [cons_sub_of_le _ (mem_powerset.mp hx)]
+#align multiset.antidiagonal_eq_map_powerset Multiset.antidiagonal_eq_map_powerset
+
+-- Porting note: Unnecessary `have` commented out to satisfy linter
```

Comment:
```suggestion
```
In this case, we can just remove the `have`. It's probably just a remnant of proof-writing, where the proof wasn't properly cleaned up.